client/components/common/tce-core/ElementDiscussion.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <v-menu
+    v-model="isVisible"
+    :close-on-content-click="false"
+    transition="slide-y-transition"
+    min-width="300"
+    attach offset-y left>
+    <template v-slot:activator="{ on: menu }">
+      <v-tooltip open-delay="800" left>
+        <template v-slot:activator="{ on: tooltip }">
+          <v-btn v-on="{ ...menu, ...tooltip }" icon>
+            <v-badge
+              v-if="unseenCommentCount"
+              :content="unseenCommentCount"
+              color="secondary" />
+            <v-icon v-else color="primary" class="pr-1">mdi-forum-outline</v-icon>
+          </v-btn>
+        </template>
+        <span>Discussion</span>
+      </v-tooltip>
+    </template>
+    <discussion
+      @save="save"
+      @update="save"
+      @remove="editorBus.emit(events.REMOVE, $event)"
+      v-bind="{ comments, user }"
+      show-heading
+      class="pa-4" />
+  </v-menu>
+</template>
+
+<script>
+import Discussion from 'tce-core/Discussion';
+import { getCommentEvents } from './utils';
+import { mapChannels } from '@/plugins/radio';
+
+export default {
+  name: 'element-discussion',
+  props: {
+    id: { type: Number, required: true },
+    uid: { type: String, required: true },
+    comments: { type: Array, required: true },
+    unseenComments: { type: Array, required: true },
```

Comment:
Lets discuss separating these two

Reply:
I removed CE unseen processor from `comment` getter. Then I passed `seen` state per element and handle unseen comments within this `tce-core` wrapper as computed. As for the `unseenCommentCount`, i need that, because I need short delay (handled in watcher) for displaying unseen comment notification. Tomorrow I will do the cleanup, and see if there's more improvements here...

Reply:
Can't the delay be created by triggering a delayed event 🤔 ?